pkg/resource/secret_reconciler.go

```diff
@@ -78,43 +80,65 @@ func NewSecretPropagatingReconciler(m manager.Manager) reconcile.Reconciler {
 			return reconcile.Result{}, errors.Wrap(IgnoreNotFound(err), errGetSecret)
 		}
 
-		// The 'to' secret is also known as the 'propagated' secret. We guard
-		// against abusers of the propagation process by requiring that both
-		// secrets consent to propagation by specifying each other's UID. We
-		// cannot know the UID of a secret that doesn't exist, so the propagated
-		// secret must be created outside of the propagation process.
-		to := &corev1.Secret{}
-		n := types.NamespacedName{
-			Namespace: from.GetAnnotations()[AnnotationKeyPropagateToNamespace],
-			Name:      from.GetAnnotations()[AnnotationKeyPropagateToName],
+		fromAnnotations := from.GetAnnotations()
+
+		uids := util.Split(fromAnnotations[AnnotationKeyPropagateToUIDs], ",")
+		names := util.Split(fromAnnotations[AnnotationKeyPropagateToNames], ",")
+		namespaces := util.Split(fromAnnotations[AnnotationKeyPropagateToNamespaces], ",")
+		propagationErrors := []error{}
+
+		if len(uids) == len(names) && len(names) == len(namespaces) {
```

Comment:
If we do stick with this comma separated annotations pattern I would prefer we returned early in the case the annotations were incongruent, rather than nesting the bulk of the logic of this controller inside this if statement.